src/core/annotation.js

```diff
@@ -50,36 +56,46 @@ class AnnotationFactory {
       subtype,
       id,
       pdfManager,
+      evaluator,
+      task,
     };
 
     switch (subtype) {
       case 'Link':
-        return new LinkAnnotation(parameters);
+        return Promise.resolve(new LinkAnnotation(parameters));
 
       case 'Text':
-        return new TextAnnotation(parameters);
+        return Promise.resolve(new TextAnnotation(parameters));
 
       case 'Widget':
         let fieldType = Util.getInheritableProperty(dict, 'FT');
         fieldType = isName(fieldType) ? fieldType.name : null;
 
+        let widget = new WidgetAnnotation(parameters);
```

Comment:
Here is a deal. Now instead of annotation itself I return promise. The reason I do it because DR dictionary should be parsed asynchronously, see also my comments below.

Also I just realized that I merged the code incorrectly for new annotations like Stamp, I'll add Promise.resolve to them as well.

Reply:
The issue here is not that that it returns a promise, but rather that a new instance is created here *unconditionally*, and most likely overwritten a few lines below. That means that this object is created unnecessarily in most cases, so if you move the creation of the object into the `default` case, you'll end up with the same behavior as before, but no intermediate object that is overwritten later.

Reply:
Ok, now I see, thank you!